### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by ivandmitrienko**
проверяй, Бро)

### CODE REVIEW

File: src/components/Checkbox.jsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  const dispatch = useDispatch();
+
+  const handleClick = () => {
+      dispatch({type: 'REMAIN TASK', task: task.id, checked:task.isChecked});
+  }
+
+  return (
+
+    <input type="checkbox" checked = {task.isChecked} name="chk" style={{ marginRight: '5px' }} onChange={handleClick}/>
+
```
Comment (DmitryMoodys) [inline src/components/Checkbox.jsx]: please, don't leave empty lines in `return` method in the end and in the beginning

File: src/components/Footer.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      youText.current.value = '';
+    }
+
+  }
+
+ 
+  
+
+
+  return (
```
Comment (DmitryMoodys) [inline src/components/Footer.jsx:21]: here, too 
too many empty lines, please, try to use max 1-2 lines to separate the logic

File: src/components/Tasks.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import Checkbox from './Checkbox';
+import './Tasks.css';
+
+export default function Tasks({ task }) {
+
+  const dispatch = useDispatch();
+
+  const deleteTask = () => {
+
+    dispatch({type: 'REMOVE TASK', task: task.id});
```
Comment (DmitryMoodys) [inline src/components/Tasks.jsx]: for the future projects:

It's good practice to move all Action Type/Name in a separate file:
export const REMOVE_TASK = 'REMOVE TASK';

https://redux.js.org/usage/reducing-boilerplate#actions

and then use as variable:
dispatch({type: REMOVE_TASK task: task.id});

File: src/components/Checkbox.jsx (hunk 1 above)
Comment (ivandmitrienko) [reply src/components/Checkbox.jsx]: Ок

File: src/components/Footer.jsx (hunk 2 above)
Comment (ivandmitrienko) [reply src/components/Footer.jsx:21]: Ok, Bro)

File: src/store/addReducer.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+const defaultState = {
+    tasks: [],
+};
+
+export function reducer(state = defaultState, action) {
+    switch (action.type) {
+        case 'ADD TASK':
```
Comment (DmitryMoodys) [inline src/store/addReducer.js]: use Action Name/Type from separate file :) 

export const REMOVE_TASK = 'REMOVE TASK';

File: src/store/addReducer.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+            return { ...state, tasks: [...state.tasks, action.task]};
+
+        case 'REMOVE TASK':
+            return { ...state, tasks: state.tasks.filter(task=>task.id!==action.task) };
+
+        case 'REMAIN TASK':
+         const indexTask = state.tasks.findIndex(task=>task.id===action.task);
+         return { ...state, tasks: [...state.tasks.slice(0,indexTask),{...state.tasks[indexTask], isChecked:!action.checked}, ...state.tasks.slice(indexTask+1)]};
```
Comment (DmitryMoodys) [inline src/store/addReducer.js]: over engineering :)  
next time lets do it in more readable way

File: src/store/addReducer.js (hunk 4 above)
Comment (ivandmitrienko) [reply src/store/addReducer.js]: Thanks 👍

File: src/store/addReducer.js (hunk 5 above)
Comment (ivandmitrienko) [reply src/store/addReducer.js]: I've got it. I'll update my code)

**Pull request merged**